File: parse_data.py
import csv
import json
import xml.etree.ElementTree as ET
import yaml


with open('data.csv', 'r') as csv_file:
    csv_reader = csv.DictReader(csv_file)
    csv_data = list(csv_reader)


xml_tree = ET.parse('data.xml')
xml_root = xml_tree.getroot()
xml_data = [{"name": person.find('name').text, "age": int(person.find('age').text), "country": person.find('country').text}
            for person in xml_root.findall('person')]


with open('data.json', 'r') as json_file:
    json_data = json.load(json_file)


with open('data.yaml', 'r') as yaml_file:
    yaml_data = yaml.safe_load(yaml_file)


print("CSV Data:", csv_data)
print("XML Data:", xml_data)
print("JSON Data:", json_data)
print("YAML Data:", yaml_data)

File: parse_data.js
const fs = require('fs');
const parse = require('csv-parse/lib/sync');
const xml2js = require('xml2js');
const yaml = require('js-yaml');


const csvData = parse(fs.readFileSync('data.csv', 'utf-8'), { columns: true });


const xmlData = [];
const xmlParser = new xml2js.Parser();
const xmlFileContent = fs.readFileSync('data.xml', 'utf-8');
xmlParser.parseString(xmlFileContent, (err, result) => {
    result.people.person.forEach(person => {
        xmlData.push({
            name: person.name[0],
            age: parseInt(person.age[0]),
            country: person.country[0]
        });
    });
});


const jsonData = JSON.parse(fs.readFileSync('data.json', 'utf-8'));


const yamlData = yaml.load(fs.readFileSync('data.yaml', 'utf-8'));


console.log("CSV Data:", csvData);
console.log("XML Data:", xmlData);
console.log("JSON Data:", jsonData);
console.log("YAML Data:", yamlData);
